src/redux/missions/missionsMiddleware.ts

```diff
@@ -26,7 +33,22 @@ export const addMissionsListener = (startListening: AppStartListening) => {
     actionCreator: onOpenConnection,
     effect: (_, api) => {
       getMissions((missions) => {
-        api.dispatch(initializeMissions(missions));
+        if (missions) {
+          api.dispatch(initializeMissions(missions));
+        }
+      });
+    }
+  });
+
+  startListening({
+    actionCreator: refreshMissions,
+    effect: (_, api) => {
+      getMissions((missions) => {
+        if (missions) {
+          api.dispatch(initializeMissions(missions));
+        } else {
+          api.dispatch(clearMissions());
+        }
```

Comment:
Why don't the openconnection and the refreshmissions have the same callback? Shouldn't they both do the same thing?